lib/scope.js

```diff
@@ -225,10 +225,36 @@ function addPattern(patternPath, bindings) {
         } else {
             bindings[pattern.name] = [patternPath];
         }
-
-    } else if (namedTypes.SpreadElement &&
-               namedTypes.SpreadElement.check(pattern)) {
-        addPattern(patternPath.get("argument"), bindings);
+    } else if (namedTypes.ObjectPattern.check(pattern)) {
+
+        patternPath.get('properties').each(function(propertyPath) {
+            var property = propertyPath.value;
+            if (namedTypes.Pattern.check(property)) {
+                addPattern(propertyPath, bindings);
+            } else  if (namedTypes.Property.check(property)) {
+                addPattern(propertyPath.get('value'), bindings);
+            } else if (namedTypes.SpreadProperty &&
+                       namedTypes.SpreadProperty.check(property)) {
+                addPattern(propertyPath.get('argument'), bindings);
+            }
+        });
+    } else if (namedTypes.ArrayPattern.check(pattern)) {
```

Comment:
Nit: `ArrayPattern` is only defined in `def/es6.js`, so can you guard this condition like so?
```js
} else if (namedTypes.ArrayPattern &&
           namedTypes.ArrayPattern.check(pattern) {
```